### PR TITLE
SONARIAC-1533 Make Kubernetes analyzer compatible with SonarLint

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/EmptyChecksVisitor.java
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/EmptyChecksVisitor.java:1]: A but strange to see such visitor :D maybe add a small comment about why?

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/KubernetesSensor.java [hunk 1]
```diff
@@ -83,6 +93,13 @@ protected void initContext(SensorContext sensorContext) {
     } else {
       LOG.debug("Skipping initialization of Helm processor");
     }
+    if (sonarLintFileListener != null) {
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/KubernetesSensor.java:96]: Why do we have a different analyzer than the one created from the `createAnalyzer` method?
Is the `createAnalyzer` method called at all in a context of SonarLint ?

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/SonarLintFileListener.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  @Override
+  public void process(ModuleFileEvent moduleFileEvent) {
+    var uri = Path.of(moduleFileEvent.getTarget().uri()).normalize().toUri().toString();
+    if (moduleFileEvent.getType() == ModuleFileEvent.Type.DELETED) {
+      projectContext.removeResource(uri);
+    } else {
+      projectContext.removeResource(uri);
+      analyzer.analyseFiles(sensorContext, List.of(moduleFileEvent.getTarget()), KubernetesLanguage.KEY);
+    }
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/SonarLintFileListener.java:70]:
```suggestion
    projectContext.removeResource(uri);
    if (moduleFileEvent.getType() != ModuleFileEvent.Type.DELETED) {
      analyzer.analyseFiles(sensorContext, List.of(moduleFileEvent.getTarget()), KubernetesLanguage.KEY);
    }
```

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/plugin/SonarLintFileListenerTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  @ParameterizedTest
+  @MethodSource
+  void shouldCallRemoveResourceAndAnalyseFilesWhenEvent(ModuleFileEvent.Type eventType) {
+    sonarLintFileListener.initContext(context, analyzer, projectContext);
+    var event = DefaultModuleFileEvent.of(inputFile2, eventType);
+
+    sonarLintFileListener.process(event);
+
+    verify(projectContext).removeResource(uri(inputFile2));
+  }
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/plugin/SonarLintFileListenerTest.java:99]: I was expecting to see a verify like `verify(analyzer).analyseFiles(...);`

File: sonar-iac-plugin/src/main/java/org/sonar/plugins/iac/IacPlugin.java [hunk 4]
```diff
@@ -35,8 +37,12 @@
 
 public class IacPlugin implements Plugin {
 
+  private static final Logger LOG = LoggerFactory.getLogger(IacPlugin.class);
+
+
   @Override
   public void define(Context context) {
+    LOG.info("AAA IacPlugin define");
```
Comment (rudy-regazzoni-sonarsource) [inline sonar-iac-plugin/src/main/java/org/sonar/plugins/iac/IacPlugin.java:45]: Is this some leftover that should be removed? :D

**Review by rudy-regazzoni-sonarsource — APPROVED** [review]
LGTM! Great job! Please see my few comments/questions/suggestions.

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/KubernetesSensor.java (hunk 1 above)
Comment (mstachniuk) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/KubernetesSensor.java:96]: The different analyzer is needed for updating `ProjectContext` when files change in SonarLint.
The old `createAnalyzer` is still used by SQ & SL contexts. 
I will add a comment to document this behavior.

File: sonar-iac-plugin/src/main/java/org/sonar/plugins/iac/IacPlugin.java (hunk 4 above)
Comment (mstachniuk) [reply sonar-iac-plugin/src/main/java/org/sonar/plugins/iac/IacPlugin.java:45]: 🤦‍♂️

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/plugin/SonarLintFileListenerTest.java (hunk 3 above)
Comment (mstachniuk) [reply iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/plugin/SonarLintFileListenerTest.java:99]: Good catch

**Pull request merged**